app/src/pages/common/settingsPage/defectTypesTab/defectTypesGroup.jsx

```diff
@@ -16,27 +22,88 @@ ColorMarker.propTypes = {
   color: PropTypes.string.isRequired,
 };
 
+@injectIntl
 export class DefectTypesGroup extends Component {
   static propTypes = {
     group: PropTypes.arrayOf(defectTypeShape).isRequired,
+    intl: intlShape.isRequired,
   };
 
+  getChartConfig() {
+    return {
+      data: {
+        columns: this.props.group.map(({ locator }) => [locator, 100]),
+        colors: this.props.group.reduce((acc, { locator, color }) => {
+          acc[locator] = color;
+          return acc;
+        }, {}),
+        type: 'donut',
+      },
+      donut: {
+        expand: false,
+        width: 12,
+        label: {
+          show: false,
+        },
+      },
+      interaction: {
+        enabled: false,
+      },
+      legend: {
+        show: false,
+      },
+      tooltip: {
+        show: false,
+      },
+      size: {
+        width: 56,
+        height: 56,
+      },
+    };
+  }
+
+  MAX_DEFECT_SUBTYPES_COUNT = 15;
+
   render() {
-    const { group } = this.props;
-
-    return group.map(({ id, longName, shortName, color }) => (
-      <div key={id} className={cx('defect-type')}>
-        <div className={cx('name-cell')}>
-          <span className={cx('defect-type-name-wrap')}>
-            <ColorMarker color={color} />
-            <span className={cx('defect-type-name')}>{longName}</span>
-          </span>
-        </div>
-        <div className={cx('abbr-cell')}>{shortName}</div>
-        <div className={cx('color-cell')}>
-          <ColorMarker color={color} />
+    const { group, intl } = this.props;
+
+    return (
+      <div>
```

Comment:
Is this div necessary? You can use `Fragment` instead

Reply:
yep, for now, let's live it as it is, I will fix it in the next PR, I've restructured the files, created subcomponents